### PR TITLE
misc cleanups

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: compiler/rustc_middle/src/mir/mod.rs [hunk 1]
```diff
@@ -1482,10 +1482,10 @@ impl SourceScope {
         let scope_data = &source_scopes[self];
         if let Some((inlined_instance, _)) = scope_data.inlined {
             Some(inlined_instance)
-        } else if let Some(inlined_scope) = scope_data.inlined_parent_scope {
-            Some(source_scopes[inlined_scope].inlined.unwrap().0)
         } else {
-            None
+            scope_data
+                .inlined_parent_scope
+                .map(|inlined_scope| source_scopes[inlined_scope].inlined.unwrap().0)
```
Comment (Zalathar) [inline compiler/rustc_middle/src/mir/mod.rs:1488]: I have mixed feelings about `manual_map` in general, but this particular pattern of replacing only the tail of an if-chain with a `map` seems like a clear downgrade to me. 

Now the final condition (using `map`) is inconsistent with the prior conditions (which are still using `if let Some` followed by `Some`), for no real benefit.

Comment (jieyouxu) [reply compiler/rustc_middle/src/mir/mod.rs:1488]: +1 I find the replacing only the tail with a `map` makes it quite hard to read

File: compiler/rustc_middle/src/ty/inhabitedness/inhabited_predicate.rs [hunk 2]
```diff
@@ -190,10 +190,8 @@ impl<'tcx> InhabitedPredicate<'tcx> {
             (Self::And(&[a, b]), c) | (c, Self::And(&[a, b])) => {
                 if let Some(ac) = a.reduce_and(tcx, c) {
                     Some(ac.and(tcx, b))
-                } else if let Some(bc) = b.reduce_and(tcx, c) {
-                    Some(Self::And(tcx.arena.alloc([a, bc])))
                 } else {
-                    None
+                    b.reduce_and(tcx, c).map(|bc| Self::And(tcx.arena.alloc([a, bc])))
                 }
```
Comment (Nadrieril) [inline compiler/rustc_middle/src/ty/inhabitedness/inhabited_predicate.rs:195]: I agree with the others, this feels less legible now than before. I won't block this though if you feel that's an improvement

**Comment by Nadrieril** [comment]
r=me when CI is green

**Pull request merged**